test/Atmos/EDMF/closures/pressure.jl

```diff
@@ -0,0 +1,58 @@
+#### Pressure model kernels
+
+"""
+    perturbation_pressure(
+        m::AtmosModel{FT},
+        press::PressureModel,
+        state::Vars,
+        diffusive::Vars,
+        aux::Vars,
+        t::Real,
+        env,
+        i,
+    ) where {FT}
+Returns the value of perturbation pressure gradient
+for updraft i, as well as the corresponding source term
+in the environmental TKE budget following He et al. (JAMES, 2020),
+given:
+ - `m`, an `AtmosModel`
+ - `press`, a `PressureModel`
+ - `state`, state variables
+ - `diffusive`, additional variables
+ - `aux`, auxiliary variables
+ - `t`, the time
+ - `env`, NamedTuple of environment variables
+ - `i`, index of the updraft
+"""
+function perturbation_pressure(
+    m::AtmosModel{FT},
+    press::PressureModel,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    env,
+    i,
+) where {FT}
+
+    # Alias convention:
+    gm = state
+    en = state.turbconv.environment
+    up = state.turbconv.updraft
+    up_a = aux.turbconv.updraft
+    up_d = diffusive.turbconv.updraft
+
+    ρinv = 1 / gm.ρ
+    N_up = n_updrafts(m.turbconv)
+    w_up = up[i].ρaw / up[i].ρa
+
+    nh_press_buoy = press.α_b * up_a[i].buoyancy
+    nh_pressure_adv = -press.α_a * w_up * up_d[i].∇w[3]
+    nh_pressure_drag =
+        press.α_d * (w_up - env.w) * abs(w_up - env.w) / press.H_up
+
+    dpdz = nh_press_buoy + nh_pressure_adv + nh_pressure_drag
+    dpdz_tke_i = up[i].ρa * (w_up - env.w) * dpdz
```

Comment:
If I understatnd it correcly, `up[i].ρa` is left out pf the pressure computation and multiplied to dpdz directly in the momentum equation? Then why do we include the `up[i].ρa` in tke pressure work instead of leaving it to the the prognostic equation?

Reply:
@yairchn ?

Reply:
Jia is correct the way the tke pressure closure is coded is wrong. it both a wrong implementation of the equation and misleading compared with how updraft pressure term.
What we should have is a single output from pressure.jl , that is dpdz_i .
and replace 
en_s.ρatke += ρaᵢ * dpdz_tke_i
with 
en_s.ρatke += ρaᵢ * (wᵢ - env.w)*dpdz_i
using the single output from pressure.jl

Reply:
Fixed.